### PR TITLE
COMDOX-857: Remove link check ignore pattern and add new glob pattern for frontmatter check

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) is related to #245 and does the following:

- Removes the link check `--ignore-pattern` from the npm script `test:links`
   - Ignoring directories is an anti-pattern and we no longer need to do it now that we have more robust validation
- Adds a new glob pattern for the frontmatter check to exclude the `src/pages/graphql/schema` directory
   - There are ~250 .md files in this directory and we've decided that the amount of effort it would take to add description metadata is not worth the potential SEO benefit.
- Fixes broken links in the `src/pages/graphql/schema/` directory that we did not catch in #245.

## Testing

1. [Failure](https://github.com/AdobeDocs/commerce-webapi/actions/runs/7010737399/job/19071878039?pr=247) - Remove `--ignore-pattern` from npm script and update frontmatter check glob pattern ([e6a89e4](https://github.com/AdobeDocs/commerce-webapi/pull/247/commits/e6a89e4949c7e76d90137da5970b95f85b612065))
   - _**NOTE:** I had to update the glob pattern in a subsequent commit (steps 3 & 5)_
1. [Success](https://github.com/AdobeDocs/commerce-webapi/actions/runs/7010885677/job/19072357633?pr=247) - Fix broken links ([78a9594](https://github.com/AdobeDocs/commerce-webapi/pull/247/commits/78a95943fc92c8fbd3ff8a430a1433bc44d85a2e))
1. [Failure](https://github.com/AdobeDocs/commerce-webapi/actions/runs/7020122507/job/19099122258?pr=247) - Introduce breaking change in other directories to test glob pattern ([22ec638](https://github.com/AdobeDocs/commerce-webapi/pull/247/commits/22ec6381459e3bcff6c0f266929996f82577f2fe))
1. [False success](https://github.com/AdobeDocs/commerce-webapi/actions/runs/7020188707/job/19099330381?pr=247) - Revert breaking changes ([82e6840](https://github.com/AdobeDocs/commerce-webapi/pull/247/commits/82e684024f76cbb699f8750c6672f53cd124f44d))
   - Mistakenly reverted correct glob pattern
1. [Success](https://github.com/AdobeDocs/commerce-webapi/actions/runs/7020212677/job/19099408519) - Apply correct glob pattern that was mistakenly reverted ([248b401](https://github.com/AdobeDocs/commerce-webapi/pull/247/commits/248b4016abcbcc060155e491285bfec419df2f5b))

### CODE REVIEW

File: .remarkrc.mjs [hunk 1]
```diff
@@ -14,9 +14,9 @@ const remarkConfig = {
 			{
 				schemas: {
 					/* One schema for many files */
-					"./.github/linters/metadata.schema.yml": [
+					'./.github/linters/metadata.schema.yml': [
```
Comment (dshevtsov) [inline .remarkrc.mjs:17]: Was this change made by IDE or on purpose?

Comment (jeff-matthews) [reply .remarkrc.mjs:17]: Not by IDE, but not necessarily on purpose. I was looking at the config example in the [plugin docs](https://github.com/JulianCataldo/remark-lint-frontmatter-schema#globally-with-patterns) and used single quotes to match. I'm happy to revert to keep consistent with other repos though.

File: src/pages/graphql/schema/b2b/purchase-order-rule/index.md [hunk 2]
```diff
@@ -80,7 +80,7 @@ The following example returns the list of purchase order approval rules.
 
 ## Get approval rule details
 
-The `purchase_order_approval_rule` query returns information about the specified approval rule. To retrieve details about the amount or quantity required to trigger an approval rule, you must specify the implementations of the [`PurchaseOrderApprovalRuleConditionInterface`](interfaces/condition.md).
+The `purchase_order_approval_rule` query returns information about the specified approval rule. To retrieve details about the amount or quantity required to trigger an approval rule, you must specify the implementations of the [`PurchaseOrderApprovalRuleConditionInterface`](https://developer.adobe.com/commerce/webapi/graphql-api/beta/index.html#definition-PurchaseOrderApprovalRuleConditionInterface).
```
Comment (keharper) [inline src/pages/graphql/schema/b2b/purchase-order-rule/index.md:83]:
```suggestion
The `purchase_order_approval_rule` query returns information about the specified approval rule. To retrieve details about the amount or quantity required to trigger an approval rule, you must specify the implementations of the [`PurchaseOrderApprovalRuleConditionInterface`](https://developer.adobe.com/commerce/webapi/graphql-api/index.html#definition-PurchaseOrderApprovalRuleConditionInterface).
```

File: src/pages/graphql/schema/cart/mutations/remove-item.md [hunk 3]
```diff
@@ -86,4 +86,4 @@ Error | Description
 `Required parameter "cart_item_id" is missing.` | The value specified in the `cart_item_id` argument is equal to zero.
 `The current user cannot perform operations on cart "XXX"` | An unauthorized user (guest) tried to remove a product from the shopping cart of authorized user (customer), or a customer tried to remove a product from the shopping cart of another customer.
 
-[Cart query output](../../cart/queries/cart.md#output-attributes) provides more information about the `Cart` object.
+[Cart query output](../../cart/queries/cart.md#input-attributes) provides more information about the `Cart` object.
```
Comment (keharper) [inline src/pages/graphql/schema/cart/mutations/remove-item.md:89]: Delete this line.

File: src/pages/graphql/schema/products/interfaces/attributes.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ Any type that implements [`ProductInterface`](https://developer.adobe.com/commer
 The `items` that are returned in a `ProductInterface` array can also contain attributes from resources external to the `CatalogGraphQl` module:
 
 -  Custom and extension attributes defined in any attribute set
--  The attribute is defined in the [PhysicalProductInterface](#physicalproductinterface) or [CustomizableOptionInterface](customizable-option.md)
+-  The attribute is defined in the [PhysicalProductInterface](https://developer.adobe.com/commerce/webapi/graphql-api/beta/index.html#definition-PhysicalProductInterface) or [CustomizableOptionInterface](customizable-option.md)
```
Comment (keharper) [inline src/pages/graphql/schema/products/interfaces/attributes.md:11]:
```suggestion
-  The attribute is defined in the [PhysicalProductInterface](https://developer.adobe.com/commerce/webapi/graphql-api/index.html#definition-PhysicalProductInterface) or [CustomizableOptionInterface](customizable-option.md)
```

File: src/pages/graphql/schema/products/interfaces/index.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ Any type that implements [`ProductInterface`](https://developer.adobe.com/commer
 The `items` that are returned in a `ProductInterface` array can also contain attributes from resources external to the `CatalogGraphQl` module:
 
 -  Custom and extension attributes defined in any attribute set
--  The attribute is defined in the [PhysicalProductInterface](#physicalproductinterface) or [CustomizableOptionInterface](customizable-option.md)
+-  The attribute is defined in the [PhysicalProductInterface](https://developer.adobe.com/commerce/webapi/graphql-api/beta/index.html#definition-PhysicalProductInterface) or [CustomizableOptionInterface](customizable-option.md)
```
Comment (keharper) [inline src/pages/graphql/schema/products/interfaces/index.md:11]:
```suggestion
-  The attribute is defined in the [PhysicalProductInterface](https://developer.adobe.com/commerce/webapi/graphql-api/index.html#definition-PhysicalProductInterface) or [CustomizableOptionInterface](customizable-option.md)
```

**Review by keharper — APPROVED** [review]

File: .remarkrc.mjs (hunk 1 above)
Comment (dshevtsov) [reply .remarkrc.mjs:17]: Never mind, I checked the file and it was already inconsistent.

**Review by dshevtsov — APPROVED** [review]

**Review by keharper — APPROVED** [review]

**Review by jhadobe — APPROVED** [review]

**Pull request merged**